src/tests/daos_perf.c

```diff
@@ -88,13 +89,79 @@ bool			 ts_nest_iterator;
 bool			ts_rebuild_only_iteration = false;
 /* rebuild without update */
 bool			ts_rebuild_no_update = false;
+/* test inside ULT */
+bool			ts_in_ult = false;
+static ABT_xstream	abt_xstream;
+
+int
+ts_abt_init(void)
+{
+	int cpuid;
+	int num_cpus;
+	int rc;
+
+	rc = ABT_init(0, NULL);
+	if (rc != ABT_SUCCESS) {
+		fprintf(stderr, "ABT init failed: %d\n", rc);
+		return -1;
+	}
+	rc = ABT_xstream_self(&abt_xstream);
+	if (rc != ABT_SUCCESS) {
+		printf("ABT get self xstream failed: %d\n", rc);
+		return -1;
+	}
+	rc = ABT_xstream_get_cpubind(abt_xstream, &cpuid);
+	if (rc != ABT_SUCCESS) {
+		fprintf(stderr, "get cpubind: %d\n", rc);
+		return -1;
+	}
+
+	rc = ABT_xstream_get_affinity(abt_xstream, 0, NULL,
+				      &num_cpus);
+	if (rc != ABT_SUCCESS) {
+		fprintf(stderr, "get num_cpus: %d\n", rc);
+		return -1;
+	}
+	cpuid = (cpuid + 1) % num_cpus;
+	rc = ABT_xstream_set_cpubind(abt_xstream, cpuid);
+	if (rc != ABT_SUCCESS) {
+		fprintf(stderr, "set affinity: %d\n", rc);
+		return -1;
+	}
```

Comment:
I think our argobots build doesn't enabled cpu affinity feature, otherwise, DAOS server xstream affinity will be changed by ABT_xstream_create_with_rank() unexpectedly.

Reply:
oh, it does need to enable affinity. (--enable-affinity) to rebuild the Argobots.  Not sure what you mean ABT_xstream_create_with_rank() ?